### ISSUE TITLE
No way to list or remove Entitlement Management Catalog Resource Roles

### ISSUE THREAD

**Issue by brolifen** [microsoftgraph/msgraph-sdk-powershell#1982]
Using **New-MgBetaEntitlementManagementAccessPackageResourceRequest** one could add resource roles to a catalog. Which is already strange due to its naming usage.

However running **Get-MgBetaEntitlementManagementAccessPackageResource** returns an error:

```
Get-MgBetaEntitlementManagementAccessPackageResource_List: Origin system is required.
```

The closest graph endpoint I could find was:
https://graph.microsoft.com/beta/identityGovernance/entitlementManagement/accessPackageAssignmentResourceRoles

And finally there is no way to remove the previously created Resource Roles.

TLDR: Adding Resource Roles is done using a badly named cmdlet, cant list or remove resource roles after creation.

**Comment by peombwa** [comment]
Thanks for following up on this.

According to the [API reference](https://learn.microsoft.com/en-us/graph/api/resources/accesspackageassignmentresourcerole?view=graph-rest-beta#methods), you can list accessPackageResourceRoles on an access package catalog by calling [`Get-MgEntitlementManagementAccessPackageCatalogAccessPackageResourceRole`](https://learn.microsoft.com/en-us/graph/api/accesspackagecatalog-list-accesspackageresourceroles?view=graph-rest-beta&tabs=powershell#example-1-retrieving-the-roles-of-a-resource-for-a-group). It does appear that the API does not support explicitly support removing accessPackageResourceRoles - https://learn.microsoft.com/en-us/graph/api/resources/accesspackageresourcerole?view=graph-rest-beta#methods. However, the API description for [create access package resource request](https://learn.microsoft.com/en-us/graph/api/entitlementmanagement-post-accesspackageresourcerequests?view=graph-rest-beta&tabs=http#request-body) says the following about removing a resource from a catalog:
> To remove a resource from a catalog, set the catalogId to be of the ID of the catalog, requestType to be AdminRemove, and the accessPackageResource the resource object to be removed. The resource object can be retrieved using [list accessPackageResources](https://learn.microsoft.com/en-us/graph/api/accesspackagecatalog-list-accesspackageresources?view=graph-rest-beta).

Please open a question for the API experts at https://developer.microsoft.com/en-us/graph/support to get help on using the API.

This repository is intended for issues related to the functionality of the module (bugs and feature requests). We may not be the best place to answer some queries that are tied to the functionality of the API or how to use an API. Please use https://developer.microsoft.com/en-us/graph/support to ask API questions.
